app/models/assignments/exam.rb

```diff
@@ -0,0 +1,108 @@
+class Exam < Assignment
+  before_create :set_exam_graders
+  before_update :update_exam_submission_times
+
+  def set_exam_graders
+    # FIXME: This is complicated, and shouldn't be here either.
```

Comment:
This is still annoyingly complicated, but probably does go here, since this is most of the logic in the class.